### ISSUE TITLE
Suggestion: include callout box on directory() usage when downstream rule requires a file within the directory

### ISSUE THREAD

**Issue by ezherman** [carpentries-incubator/snakemake-novice-bioinformatics#7]
When using Snakemake for my own work, I encountered the scenario where a program outputs to a directory, so I used `directory()` in the output. However, a downstream rule required one of the files created within that directory as the input. The only way I found to show Snakemake that the rules are linked, was to include the relevant file as one of the outputs in the upstream rule (even though the file does not feature in the shell script of that rule).

If there is a more elegant solution to this type of scenario, it might be good to include it as a callout box. I imagine that this is something learners will encounter in their own applications of Snakemake.

**Comment by tbooth** [comment]
Interesting suggestion. My personal feeling is that there are various situations like this that occur in Snakemake and there are always going to be some requirements and patterns that don't belong in a beginner course, so I'm wary of putting in too many callout tips that could be a distraction to beginners trying to grasp the basic concepts.

In terms of this particular scenario, I'd think that either having the downstream rule depending on the whole directory or else declaring the extra file output, as you did, will work. Or if all you wanted from the first rule was that file, maybe you actually want to throw the rest of the output away. Which option is better depends on the exact workflow - often I find in Snakemake there are two or more options and both are equally workable.

**Comment by ezherman** [comment]
Hi @tbooth, makes sense regarding the extra callout box. And thanks for your thoughts on this scenario! 

I suggested this addition thinking there may be a more elegant way for Snakemake to recognise that an input file is part of a directory that is the output of another rule (i.e. without having to specify the directory additionally as an input). But it sounds like the way I did it is acceptable, so that's good to know!
